core/src/package.rs

```diff
@@ -45,6 +50,62 @@ pub enum PackageImplementation {
     Native,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub enum ArgType {
+    #[serde(alias = "string")]
+    String,
+    #[serde(alias = "int", alias = "integer", alias = "i64")]
+    Integer,
+    #[serde(
+        rename = "Unsigned integer",
+        alias = "uint",
+        alias = "unsigned_integer",
+        alias = "u64"
+    )]
+    UnsignedInteger,
```

Comment:
Hmm, this is mostly added as a simple validation step, to ensure that the integer is non-negative. Let's say that we take an argument `repeat` which controls how many times to repeat a statement, it would clearly not make any sense to have a negative number there. There are two options to handle this; either declare it as an `integer`, make a check that it is non-negative and if it is, fail in some way, or declare it as an `unsigned_integer`, and thus have the assurance that it is in fact non-negative. The benefit with having unsigned integer as its own type is that in many of the cases (in fact, all cases in our standard packages) we don't accept negative numbers, and if Core already knows that a package expects a non-negative number, it can give a more appropriate error message without launching the package itself. Also, it helps if the package is poorly documented or straight up expects to be able to parse the integer as an unsigned integer and crashes otherwise, which wouldn't give very good error messages.

I don't really understand where the JSON Schema comes in. Do we want to convert a manifest to a JSON Schema? In that case, we can probably encode `UnsignedInteger` as `type: integer, minimum: 0`. Or do you think we should accept manifests in the JSON Schema format? In that case, we can just parse `integer` as `integer`, and never generate the UnsignedInteger variant. Or do you mean that packages will use JSON Schema to parse its arguments? In that case, that package may choose to accept `integer` instead, or choose to accept `unsigned_integer` and parse it into the JSON Schema type `integer`.

Of course, we can remove this type if it is inappropriate, but I do think that it has nice benefits. If you would explain how you think JSON Schema comes into the picture, I would be glad to hear, or if you have any other opinions, feel free to share them!